lib/fastifySession.js

```diff
@@ -78,25 +84,23 @@ function session (fastify, options, next) {
         })
         return
       }
-      request.session = new Session(
-        request,
-        idGenerator,
-        cookieOpts,
-        cookieSigner,
-        session,
-        decryptedSessionId
-      )
+
+      request.session = restoredSession
       done()
     })
   }
 
+  function getCookieSessionId (request) {
```

Comment:
We could improve this:

```js
  const getCookieSessionId = hasCookiePrefix
    ? function getCookieSessionId(request) {
      const cookieValue = request.cookies[cookieName]
      return (
        cookieValue &&
        cookieValue.startsWith(cookiePrefix) &&
        cookieValue.slice(cookiePrefixLength)
      )
    }
    : function getCookieSessionId(request) {
      return request.cookies[cookieName]
    }

```

Reply:
Should be always faster when we dont use a cookiePrefix

Reply:
What about this?

Reply:
Oops sorry missed it